spark/src/test/scala/org/apache/spark/sql/delta/DeltaUpdateCatalogSuiteBase.scala

```diff
@@ -0,0 +1,312 @@
+      df2.writeTo(tbl)
+        .option("mergeSchema", "true")
+        .append()
+
+      val schema = new StructType()
+        .add("id", LongType)
+        .add("part", DoubleType)
+        .add("str", new StructType()
+          .add("int", IntegerType)
+          .add("id2", LongType)) // New columns go to the end
+      verifyTableMetadataAsync(expectedSchema = schema)
+    }
+  }
+
+
+  test("merge") {
+    val tmp = "tmpView"
+    withDeltaTable { df =>
+      withTempView(tmp) {
+        withSQLConf(DeltaSQLConf.DELTA_SCHEMA_AUTO_MIGRATE.key -> "true") {
+          df.withColumn("id2", 'id).createOrReplaceTempView(tmp)
+          sql(
+            s"""MERGE INTO $tbl t
+               |USING $tmp s
+               |ON t.id = s.id
+               |WHEN NOT MATCHED THEN INSERT *
+             """.stripMargin)
+
+          verifyTableMetadataAsync(df.withColumn("id2", 'id).schema.asNullable)
+        }
+      }
+    }
+  }
+
+  test("creating and replacing a table puts the schema and table properties in the metastore") {
+    withTable(tbl) {
+      val df = spark.range(10).withColumn("part", 'id / 2).withColumn("id2", 'id)
+      df.writeTo(tbl)
+        .tableProperty("delta.checkpointInterval", "5")
+        .tableProperty("some", "thing")
+        .partitionedBy('part)
+        .using("delta")
+        .create()
+
+      verifyTableMetadata(
+        expectedSchema = df.schema.asNullable,
+        expectedProperties = getBaseProperties(snapshot) ++ Map(
+          "delta.checkpointInterval" -> "5",
+          "some" -> "thing")
+      )
+
+      val df2 = spark.range(10).withColumn("part", 'id / 2)
+      df2.writeTo(tbl)
+        .tableProperty("other", "thing")
+        .using("delta")
+        .replace()
+
+      verifyTableMetadata(
+        expectedSchema = df2.schema.asNullable,
+        expectedProperties = getBaseProperties(snapshot) ++ Map("other" -> "thing")
+      )
+    }
+  }
+
+  test("creating table in metastore over existing path") {
+    withTempDir { dir =>
+      withTable(tbl) {
+        val df = spark.range(10).withColumn("part", 'id % 2).withColumn("id2", 'id)
+        df.write.format("delta").partitionBy("part").save(dir.getCanonicalPath)
+
+        sql(s"CREATE TABLE $tbl USING delta LOCATION '${dir.getCanonicalPath}'")
+        verifyTableMetadata(df.schema.asNullable)
+      }
+    }
+  }
+
+  test("replacing non-Delta table") {
+    withTable(tbl) {
+      val df = spark.range(10).withColumn("part", 'id / 2).withColumn("id2", 'id)
+      df.writeTo(tbl)
+        .tableProperty("delta.checkpointInterval", "5")
+        .tableProperty("some", "thing")
+        .partitionedBy('part)
+        .using("parquet")
+        .create()
+
+      val e = intercept[AnalysisException] {
+        df.writeTo(tbl).using("delta").replace()
+      }
+
+      assert(e.getMessage.contains("not a Delta table"))
+    }
+  }
+
+  test("alter table add columns") {
```

Comment:
Can we also check that the truncation of schemas with long nested fields works with converted tables?